x-pack/plugins/actions/server/builtin_action_types/servicenow/utils.ts

```diff
@@ -69,3 +82,129 @@ export const throwIfSubActionIsNotSupported = ({
     throw new Error(errorMessage);
   }
 };
+
+export interface GetAccessTokenAndAxiosInstanceOpts {
+  connectorId: string;
+  logger: Logger;
+  configurationUtilities: ActionsConfigurationUtilities;
+  credentials: ExternalServiceCredentials;
+  snServiceUrl: string;
+  connectorTokenClient: ConnectorTokenClientContract;
+}
+
+export const getAxiosInstance = ({
+  connectorId,
+  logger,
+  configurationUtilities,
+  credentials,
+  snServiceUrl,
+  connectorTokenClient,
+}: GetAccessTokenAndAxiosInstanceOpts): AxiosInstance => {
+  const { config, secrets } = credentials;
+  const { isOAuth } = config as ServiceNowPublicConfigurationType;
+  const { username, password } = secrets as ServiceNowSecretConfigurationType;
+
+  let axiosInstance;
+
+  if (!isOAuth && username && password) {
+    axiosInstance = axios.create({
+      auth: { username, password },
+    });
+  } else {
+    axiosInstance = axios.create();
+    axiosInstance.interceptors.request.use(
+      async (axiosConfig: AxiosRequestConfig) => {
+        const accessToken = await getAccessToken({
+          connectorId,
+          logger,
+          configurationUtilities,
+          credentials: {
+            config: config as ServiceNowPublicConfigurationType,
+            secrets,
+          },
+          snServiceUrl,
+          connectorTokenClient,
+        });
+        axiosConfig.headers.Authorization = accessToken;
+        return axiosConfig;
+      },
+      (error) => {
+        Promise.reject(error);
+      }
```

Comment:
__question/concern:__ hey @ymao1, while upgrading axios in https://github.com/elastic/kibana/pull/111655 noticed two weird things in this code:

* `accessToken` can be `null`, what do you want to do in this case? Do you want to throw? Do you want to remove `Authorization` header from ` axiosConfig.headers` if it's present already or something else?
* The `catch` handler doesn't seem to do anything except for swallowing all errors since rejected promise isn't actually returned from the `catch` callback. Is it intentional?

Reply:
@azasypkin Thanks for pinging. These are good catches! If the access token is null, we should probably throw to avoid making unnecessary requests to ServiceNow and yes, the `Promise.reject` should be returned 🙈 . I am about to start working on this issue: https://github.com/elastic/kibana/issues/131231 so I can make the changes in that PR or in a separate one if you need it faster?

Reply:
> I am about to start working on this issue: https://github.com/elastic/kibana/issues/131231 so I can make the changes in that PR or in a separate one if you need it faster?

No rush here, it's not blocking anything, just noticed this while making TS compiler happy with the new Axios types. Feel free to deal with this in https://github.com/elastic/kibana/issues/131231, thanks!